proposals/using-alias-types.md

```diff
@@ -0,0 +1,52 @@
+# Allow using alias directive to reference any kind of Type
+
+## Summary
+Relax the [using_alias_directive](https://github.com/dotnet/csharplang/blob/main/spec/namespaces.md#using-alias-directives) to allow it to point at any sort of type, not just named types.  This would support types not allowed today, like: tuple types, pointer types, array types, etc.  For example, this would now be allowed:
+
+```c#
+using Point = (int x, int y);
+```
+
+## Motivation
+For ages, C# has had the ability to introduce aliases for namespaces and named types (classes, delegated, interfaces, records and structs).  This worked acceptably well as it provided a means to introduce non-conflicting names in cases where a normal named pulled in from `using_directive`s might be ambiguous, and it allowed a way to provide a simpler name when dealing with complex generic types.  However, the rise of additional complex type symbols in the language has caused more use to arise where aliases would be valuable but are currently not allowed.  For example, both tuples and function-pointers often can have large and complex regular textual forms that can be painful to continually write out, and a burden to try to read.  Aliases would help in these cases by giving a short, developer-provided, name that can then be used in place of those full structural forms.
+
+## Detailed design
+We will change the grammar of `using_alias_directive` thusly:
+
+```
+using_alias_directive
+-    : 'using' identifier '=' namespace_or_type_name ';'
++    : 'using' identifier '=' (namespace_name | type) ';'
```

Comment:
Does `type` include top-level nullability annotation?
I assume this allows built-in types `using Id = int;`. Is that right?

Reply:
> Does type include top-level nullability annotation?

Yes.  It would.

> I assume this allows built-in types using Id = int;. Is that right?

yes, it would.

Reply:
> Yes. It would [include top-level nullability annotation].

From our offline discussion, this raises a question for how the three annotations (`?`, `!` and `~`/oblivious) interact in the declaration versus the use-site. For example, what happens if you put `?` in the declaration and the use-site?